Makefile

```diff
@@ -1,18 +1,36 @@
-install: testcert.cert testcert.key
-	pipenv install --system --deploy
+# Path to python binary
+PYTHON ?= $(shell which python3)
 
-test: testcert.cert testcert.key
+# Path to virtual environment
+VIRTUAL_ENV ?= '.venv'
+
+$(VIRTUAL_ENV): $(PYTHON)
+	virtualenv $(VIRTUAL_ENV) --python=$(PYTHON)
+
+.PHONY: install
+install: $(VIRTUAL_ENV)
+	. $(VIRTUAL_ENV)/bin/activate; \
+		pip install pipenv==2018.11.26; \
+		pipenv install --deploy
+
+.PHONY: test
+test: testcert.cert testcert.key install
+	. $(VIRTUAL_ENV)/bin/activate; \
 	./test.py -s
 
+.PHONY: docker-build
```

Comment:
we should probably merge 'build' into this file. no reason to have two entrypoints...

Reply:
Will do